units/provisioning_create_host.go

```diff
@@ -144,14 +145,50 @@ func (j *createHostJob) Run(ctx context.Context) {
 			grip.Error(message.WrapError(err, message.Fields{
 				"host_id":  j.HostID,
 				"attempt":  j.CurrentAttempt,
-				"distro":   j.host.Distro,
+				"distro":   j.host.Distro.Id,
 				"job":      j.ID(),
 				"provider": j.host.Provider,
 				"message":  "could not remove intent document",
 				"outcome":  "host pool may exceed maxhost limit",
 			}))
 
 			return
+		} else if !j.host.IsSubjectToHostCreationThrottle() {
+			// pass (spawn hosts and friends should always
+			// try and start)
+		} else if j.host.Status == evergreen.HostBuilding {
+			// pass (containers or similar that are
+			// building should continue to provision)
+		} else {
+			var (
+				numProv            int
+				runningHosts       int
+				distroRunningHosts int
+			)
+
+			numProv, err = host.CountProvisioning()
+			if err != nil {
+				j.AddError(errors.Wrap(err, "problem getting count of pending pool size"))
+				return
+			}
+			distroRunningHosts, err = host.CountRunningHosts(j.host.Distro.Id)
+			if err != nil {
+				j.AddError(errors.Wrap(err, "problem getting count of pending pool size"))
+				return
+			}
+
+			runningHosts, err = host.CountRunningHosts("")
```

Comment:
This includes both static and hosts that aren't up, which I think you want to exclude since it will make us hit the passthrough below a lot more than we think we're hitting it